service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -50,6 +182,47 @@ public ResponseEntity<MethodListResponse> getMethods(
     return ResponseEntity.ok(new MethodListResponse().methods(methodDetails));
   }
 
+  public static List<String> validateMethod(PostMethodRequest methodRequest) {
+    String methodName = methodRequest.getMethodName();
+    String methodSource = methodRequest.getMethodSource();
+    String methodVersion = methodRequest.getMethodVersion();
+    String methodUrl = methodRequest.getMethodUrl();
+    List<String> errors = new ArrayList<>();
+
+    if (methodName == null || methodName.trim().isEmpty()) {
+      errors.add("method_name is required");
+    }
+
+    if (methodSource == null || methodSource.trim().isEmpty()) {
+      errors.add("method_source is required");
+    } else {
+      // verify that method source is supported. Currently, we only support GitHub as the source
+      if (SUPPORTED_SOURCES.stream().noneMatch(methodSource::equalsIgnoreCase)) {
+        errors.add("method_source is invalid. Supported source(s): " + SUPPORTED_SOURCES);
+      }
+    }
+
+    if (methodVersion == null || methodVersion.trim().isEmpty()) {
+      errors.add("method_version is required");
+    }
+
+    if (methodUrl == null || methodUrl.trim().isEmpty()) {
+      errors.add("method_url is required");
+    } else {
+      // verify that URL is valid, and it's host is supported
+      try {
+        URI uri = new URI(methodUrl);
```

Comment:
I think you can do a lot of these required fields and format validations in `openapi` yaml. If you do that, our spring integration makes the values come in as URIs automatically without this manual string conversion

Reply:
Interesting. I had no idea. I will look into that

Reply:
@cjllanwarne It seems that if properties are marked required in openapi.yaml and then not passed in the request body are received as null in the java code. So non-null checks might still be needed. For the `method_url` I tried adding `format: uri` but that doesn't check if the value is a URI or not. Am I missing something?